### DISCUSSION TITLE
Nuxt Roadmap showing Nuxt 4 being in the first quarter of 2024

### DISCUSSION THREAD

**Discussion by freddiv** [nuxt/nuxt#25242] [Questions]
I am currently starting the process of rebuilding a large Nuxt 2 app that will probably take us a year to migrate (Nuxt-bridge updating is incompatible with our app as soon as I try switching on any of the Nuxt3 features the application won't build). I do not want to build on a a framework that will be being end-dated in the next couple of years. Can anyone tell me what the expected run of Nuxt 3 will be after Nuxt 4 is introduced? There is currently one UI developer on this project and am also tasked with upgrading 4-5 other smaller but still pretty substantial applications. There are specific agency directives to remove any security issues and currently Nuxt 2 cannot be upgraded for many of the packages in our apps. Nuxt 3 looks good but may well end up in the same situation as v2 after a year of not getting updated releases.

**Comment by masyoudi** [comment]
same question [here](https://github.com/nuxt/nuxt/discussions/25182)

**Comment by TheAlexLichter** [comment]
@freddiv Nuxt 3 -> Nuxt 4 will be way smaller and fine-grained changes, no comparison to Nuxt 2 -> Nuxt 3. So no worries, upgrading to Nuxt 3 is the right direction

**Comment by carbdias** [comment]
I agree, moreover, most of the work in migrating from Nuxt 2 -> Nuxt 3, is the migration of Vue 2 to Vue 3, which have real syntaxis changes, to composition API. So, I believe from Nuxt 3 to Nuxt 4, Vue version will maintain in vue 3, so no major changes will happen. (correct me if I'm wrong)

**Comment by samuelkitazume** [comment]
@manniL when can we expect to have some sort of preview on what should be changing and where would be the breaking changes?

**Reply by TheAlexLichter** [reply]
Here is [a list of all issues & PRs targeting 4.x](https://github.com/nuxt/nuxt/issues?q=sort%3Aupdated-desc+label%3A4.x+). This should help as "preview". Be aware that it doesn't mean that all of the issues will actually land in 4.x (or ever land) ☺️
